### PR TITLE
adding tracking with correlationid

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

adding tracking with correlationid
## References

* #203

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        private readonly ICorrelationContext _correlationContext;
+
+        public CorrelationIdMiddleware(
+            ICorrelationContext correlationContext)
+        {
+            _correlationContext = correlationContext;
+        }
+
+        public async Task Invoke(FunctionContext context, [NotNull] FunctionExecutionDelegate next)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs:35]: think you can skip the `[NotNull]` attribute

Comment (xnetsrak) [reply source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs:35]: Yes, but then I need 
            if (next == null) throw new ArgumentNullException(nameof(next));
@michaelmajgaard

**Review by michaelmajgaard — DISMISSED** [review]

File: source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs (hunk 1 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs:35]: I know, but we already use that pattern for `context`

Comment (xnetsrak) [reply source/Energinet.DataHub.PostOffice.Common/CorrelationIdMiddleware.cs]: You are Lead :)

**Review by defectiveAi — APPROVED** [review]

File: source/Energinet.DataHub.PostOffice.Infrastructure/Correlation/TraceContext.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /// <summary>
+    /// Implementation of w3c trace context
+    /// </summary>
+    /// <remarks>
+    /// For now, the implementation doesn't handle all validation nor uses version or flags.
+    /// Specification can be found here: https://www.w3.org/TR/trace-context/#trace-id
+    /// </remarks>
+    public class TraceContext
+    {
+        // TODO: Use ActivityContext.Parse()?
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.Infrastructure/Correlation/TraceContext.cs:28]: Is this forgotten copy/pasta?

**Review by michaelmajgaard — APPROVED** [review]

File: source/Energinet.DataHub.PostOffice.Infrastructure/Correlation/TraceContext.cs (hunk 2 above)
Comment (xnetsrak) [reply source/Energinet.DataHub.PostOffice.Infrastructure/Correlation/TraceContext.cs:28]: its copy / pasta :)

**Pull request merged**